tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/routes/DownloadAllRoutes.scala

```diff
@@ -156,24 +159,32 @@ class DownloadAllRoutes @Inject()(@Named(InjectionKeys.RFC_8621) val authenticat
     val id: String = request.param(emailIdParam)
     SMono.fromCallable(() => messageIdFactory.fromString(id: String))
       .onErrorResume(e => SMono.error(MessageNotFoundException(id, e)))
-      .flatMap(messageId => SFlux(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS_WITH_ATTACHMENTS_METADATA, mailboxSession))
+      .flatMap(messageId => SFlux(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.FULL_CONTENT, mailboxSession))
         .singleOrEmpty()
-        .switchIfEmpty(SMono.error(MessageNotFoundException(id))))
-      .flatMapMany(messageResult => SFlux.fromIterable(messageResult.getLoadedAttachments.asScala))
-      .flatMap(attachment => getBlob(attachment.getAttachmentId, mailboxSession)
-        .map(blob => BlobWithName(blob, attachment.getName.orElse(DEFAULT_FILE_NAME))))
-      .collectSeq()
-      .flatMap(blobs => downloadBlobs(
-        optionalName = queryParam(request, nameParam),
-        response = response,
-        blobs)
-        .`then`())
+        .switchIfEmpty(SMono.error(MessageNotFoundException(id)))
+        .flatMapMany(messageResult => getAttachments(messageResult))
+        .flatMap(attachment => getBlob(attachment, messageId)
+          .map(blob => BlobWithName(blob, attachment.getName.orElse(DEFAULT_FILE_NAME))))
+        .collectSeq()
+        .flatMap(blobs => downloadBlobs(
+            optionalName = queryParam(request, nameParam),
+            response = response,
+            blobs)
+          .`then`()))
   }
 
-  private def getBlob(attachmentId: AttachmentId, mailboxSession: MailboxSession): SMono[Blob] =
-    SMono(attachmentManager.getAttachmentReactive(attachmentId, mailboxSession))
-      .flatMap(attachmentMetadata => SMono(attachmentManager.loadReactive(attachmentMetadata, mailboxSession))
-        .map(content => AttachmentBlob(attachmentMetadata, content)))
+  private def getAttachments(messageResult: MessageResult): SFlux[ParsedAttachment] =
+    SFlux.fromIterable(messageParser.retrieveAttachments(messageResult.getFullContent.getInputStream)
+      .getAttachments
+      .asScala)
+
+  private def getBlob(attachment: ParsedAttachment, messageId: MessageId): SMono[Blob] = // blobResolver?
```

Comment:
```suggestion
  private def getBlob(attachment: ParsedAttachment, messageId: MessageId): SMono[Blob] = 
```

Reply:
Sorry I forced push as I did fix it in a commit and was lazy to rebase :)